### PR TITLE
[RPA-1483] Check and update resources by tenantId

### PR DESCRIPTION
**Pull request by marcin-depcik**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-scheduler/src/scheduler-database/process-instance/process-instance.controller.ts [hunk 1]
```diff
@@ -36,21 +36,24 @@ export class ProcessInstanceController {
...
+    ) {
         return this.processInstanceService.getOne(id, user);
     }
 
     @Get(':id/subprocesses/GetPage')
     getSubprocesses(
         @Param('id') id: ProcessInstance['id'],
         @Pageable() paging: Paging,
-        @UserDecorator() user: User
+        @UserDecorator() user: User,
```
Comment (star-in-the-jar) [inline runbotics/runbotics-scheduler/src/scheduler-database/process-instance/process-instance.controller.ts:56]: do you have some extension or is it ESlint?

Comment (star-in-the-jar) [reply runbotics/runbotics-scheduler/src/scheduler-database/process-instance/process-instance.controller.ts:56]: that adds comma

File: runbotics/runbotics-scheduler/src/scheduler-database/tags/tag.service.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export class TagService {
         private readonly tagRepository: Repository<Tag>
     ) {}
 
-    async getAll(tenantId: string, searchPhrase?: string) {
+    async getAll(tenantId: string, searchPhrase: string | undefined) {
```
Comment (star-in-the-jar) [inline runbotics/runbotics-scheduler/src/scheduler-database/tags/tag.service.ts]: what is the reason? shouldn't it stay as it was?

**Review by star-in-the-jar — APPROVED** [review]

File: runbotics/runbotics-scheduler/src/scheduler-database/process-instance/process-instance.controller.ts (hunk 1 above)
Comment (marcin-depcik) [reply runbotics/runbotics-scheduler/src/scheduler-database/process-instance/process-instance.controller.ts:56]: indeed, my hand is an extension :D

File: runbotics/runbotics-scheduler/src/scheduler-database/tags/tag.service.ts (hunk 2 above)
Comment (marcin-depcik) [reply runbotics/runbotics-scheduler/src/scheduler-database/tags/tag.service.ts]: so far I know, `?` means that argument is optional, so it is unnecessary to pass to the function, but `string | undefined` means that argument must be passed but can be undefined

File: runbotics/runbotics-scheduler/src/scheduler-database/process-instance/process-instance.controller.ts (hunk 1 above)
Comment (marcin-depcik) [reply runbotics/runbotics-scheduler/src/scheduler-database/process-instance/process-instance.controller.ts:56]: but I think you can configure prettier in the same way

File: runbotics/runbotics-scheduler/src/queue/queue.service.ts [hunk 3]
```diff
@@ -169,10 +169,11 @@ export class QueueService implements OnModuleInit {
     }
 
     async validateProcessAccess({ process, user, triggered = false }: ValidateProcessAccessProps) {
-        const hasAccess = process.createdBy?.id === user?.id;
-        const isPublic = process.isPublic;
+        const hasTenantAccess = process.tenantId === user.tenantId;
+        const hasAccess = hasTenantAccess && process.createdBy?.id === user?.id;
```
Comment (bhszewczyk) [inline runbotics/runbotics-scheduler/src/queue/queue.service.ts]: what do you think about renaming this to hasProcessAccess?

File: runbotics/runbotics-scheduler/src/scheduler-database/process-instance/process-instance.controller.ts (hunk 1 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-scheduler/src/scheduler-database/process-instance/process-instance.controller.ts:56]: woooooow, nice
image

File: runbotics/runbotics-scheduler/src/scheduler-database/tags/tag.service.ts (hunk 2 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-scheduler/src/scheduler-database/tags/tag.service.ts]: correct, I just wonder why is it required

Comment (star-in-the-jar) [reply runbotics/runbotics-scheduler/src/scheduler-database/tags/tag.service.ts]: I think we should pass it to the function only if it is not undefined

File: runbotics/runbotics-scheduler/src/scheduler-database/tenant/tenant.controller.ts [hunk 4]
```diff
@@ -34,11 +34,13 @@ export class TenantController {
 
     @GetWithTenant('me')
     @FeatureKeys(FeatureKey.TENANT_READ)
-    async getTenantByUser(@UserDecorator('tenantId') id: Tenant['id']) {
-        const tenant = await this.tenantService.getById(id);
+    async getTenantByUser(
+        @UserDecorator('tenantId') tenantId: User['tenantId'],
```
Comment (bhszewczyk) [inline runbotics/runbotics-scheduler/src/scheduler-database/tenant/tenant.controller.ts]: you could make these two consistent 1. @UserDecorator('tenantId') tenantId: User['tenantId']  2. @UserDecorator() { tenantId }: User
personally, I prefer @UserDecorator() { tenantId }: User,

**Review by bhszewczyk — APPROVED** [review]

**Pull request merged**